### PR TITLE
Add conditions for manufacturer fields

### LINKED ISSUE #32889: FO-Cart content still displayed even deleted until you refresh the page
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments
Hello,

When you try to delete an item from your card, visually nothing happens,in fact the item still displayed until you click refresh button it disappears. 
This issue happens only when you choose second language for your shop (Not the default language).

See attached screen record:(Tested with French as a second language)

https://github.com/PrestaShop/PrestaShop/assets/92626242/4eea73b9-2075-4169-bf01-ca093852dd26

Thanks!
### Expected behavior

When we click on delete button, the items list on the card should be updated right on click and the item selected should be removed no matter what the language selected.

### Steps to reproduce

1. Go to FO
2. Choose an article > click on ADD TO CART button
3. Click on PROCEED TO CHECKOUT 
4. In the displayed SHOPPING CART click on delete button for any item
5. See the bug : Nothing happens : The item still there

### PrestaShop version(s) where the bug happened

8.0.x & develop

### PHP version(s) where the bug happened

7.4 & 8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      |  The "id_manufacturer" property was missing in a call from the Google Analytics module, resulting in an error of type 'undefined key "id_manufacturer"' when the dev mode was enabled. <br/> It was necessary to be in multi-languages to obtain the error, as previously the content was cached since the cache key included the language ID.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | see #32889
| Fixed ticket?     | Fixes #32889
| Related PRs       | -
| Sponsor company   | -

### CODE REVIEW

**Review by matks — DISMISSED** [review]

File: classes/Product.php
Comment (kpodemski) [inline classes/Product.php]:
```suggestion
        if (empty($row['manufacturer_name']) && !empty((int) $row['id_manufacturer'])) {
            $row['manufacturer_name'] = Manufacturer::getNameById((int) $row['id_manufacturer']);
        } else {
            $row['manufacturer_name'] = null;
        }
```

Comment (kpodemski) [reply classes/Product.php]: that should do it, right?

Comment (M0rgan01) [reply classes/Product.php]: it also works, in addition to gaining readability. I modify my copy

**Review by kpodemski — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Review by sallemiines — COMMENTED** [review]
Hello @M0rgan01  
Thanx for the PR ! 

I didn't manage to reproduce the issue on `80x` ! without PR the functionality is OK ! 
you can see the attached screen recorder : 

**PS80X WITHOUT PR** 

https://github.com/PrestaShop/PrestaShop/assets/31663369/247a891d-95d1-4a42-ae7c-974ed719a16a


But on`81x` and `develop` the issue exist : 

https://github.com/PrestaShop/PrestaShop/assets/31663369/ad51f011-61d7-4e78-91b7-f3811e488cd0

So in this case , do we close the PR ? or you can change the base branch from 80x to develop/81x ? 

Waiting for your feedback ! 
Thank you

**Comment by kpodemski** [comment]
We should agree to merge it. It's no brainer that it is a valid fix.

**Comment by M0rgan01** [comment]
@sallemiines You have to be in multi-language and in dev mode to reproduce the problem

**Review by MhiriFaten — COMMENTED** [review]
hello @M0rgan01 ,

The issue does not exist on 8.0.x from the beginning, I reproduced it on 8.1.x and on develop.
Could you recheck it please .?

**Comment by M0rgan01** [comment]
@MhiriFaten I confirm that I do not reproduce the "error" in 8.0.x, I think that we have additional checks on 8.1.x which stop processing in dev mode. This can be done in 8.0.x because there is a "silent" warning

**Review by matthieu-rolland — DISMISSED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @M0rgan01 ,

I have reproduced the issue. But it appears only randomly. 

https://github.com/PrestaShop/PrestaShop/assets/16019289/d09c5683-67dd-41b1-bd66-e04202fe2344

I tested your PR, the product can always be deleted from the cart. Tested with multiple types of products ✅ 

Link to auto tests : https://github.com/florine2623/testing_pr/actions/runs/5531847274 :x:

`Catalog - Brands & suppliers` part is broken

⚠️ Please wait for @PrestaShop/qa-automation before merging this PR.

Thanks!

**Comment by nesrineabdmouleh** [comment]
Hello @florine, 

A 500 internal server error is displayed in Catalog > Brands & suppliers page, could you check it please ?

Capture d’écran de 2023-07-13 09-58-28

Thank you!

**Comment by florine2623** [comment]
Hello @nesrineabdmouleh ,

On my end, I have no error. I can access the page and add a Brand or a Supplier as expected.

https://github.com/PrestaShop/PrestaShop/assets/16019289/e6f267a0-736b-4e05-8804-2ac54bd644d3

**Comment by nesrineabdmouleh** [comment]
Hello @florine2623,

Great! 
I didn't install the PR on my local to reproduce, but I checked the screenshot from the run, please restart the failed job!
If UI tests are :heavy_check_mark: you can approve :)

**Review by florine2623 — DISMISSED** [review]
The tests are green : https://github.com/florine2623/testing_pr/actions/runs/5531847274

It is QA ✅ !

**Review by Progi1984 — CHANGES_REQUESTED** [review]
The `8.0.x` isn't locked for release of 8.0.5 ?

**Review by Progi1984 — DISMISSED** [review]

**Comment by M0rgan01** [comment]
test: https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/5612478439

**Review by boherm — DISMISSED** [review]

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @M0rgan01 ,

Tested the PR on `8.1.x`.

Sorry but it is NOK this time 😓 

Sometimes I can, sometimes I can't delete products from the cart.
It seems like there's no specific rule.

I tested with multiples types of products, different quantities, different languages.

https://github.com/PrestaShop/PrestaShop/assets/16019289/ea5c7a22-a1b0-4c00-9751-b44598c993a7

**Review by tleon — DISMISSED** [review]

**Review by Hlavtox — DISMISSED** [review]

**Comment by M0rgan01** [comment]
I went back to the code I originally proposed, with the call to isset. It verifies the presence of the key that is causing the problem. Sorry for the multiple reviews :sweat_smile:

**Review by kpodemski — APPROVED** [review]
I'm not sure how using isset first and !empty later supposed to fix the recorded behavior, but I'm accepting it anyway :D 

If you didn't do it this time, the next time please consult with QA, so you can, for example, see with them Network tab and see what are the errors under the hood. We don't know if error from the recording comes from the manufacturer

**Review by Hlavtox — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Comment by M0rgan01** [comment]
@kpodemski That's word for word, exactly what happened with the QA. I checked the @florine2623  network tab, and the error persisted, even with the previous fix. After redoing several tests, it actually lacks an isset.

**Comment by kpodemski** [comment]
ok, great, thanks @M0rgan01 👍🏻 we'll never stop learning :D

**Review by florine2623 — APPROVED** [review]
Hello @M0rgan01 ,

Tested it with the latest changes !

Tested with multiple types of products, multiple browsers, languages, logged-in and logged out of customer account.
LGTM, products can always be deleted from cart. 

QA ✅

**Comment by kpodemski** [comment]
thank you @M0rgan01

**Comment by Hlavtox** [comment]
This broke the logic. :(

If there is something in `manufacturer_name`, it gets assigned to null.

**Pull request merged**
